File: mobile/src/main/java/net/stecky/puzzlemethis/events/behaviors/BadBehavior.java
package net.stecky.puzzlemethis.events.behaviors;

import java.time.ZonedDateTime;

/**
 * Created by sb185219 on 11/9/2017.
 */

public class BadBehavior extends Behavior
{
    private int severity;
    private String duration;

    public BadBehavior(BehaviorSubType type)
    {
        super(type);
    }

    public BadBehavior(BehaviorSubType type, ZonedDateTime timestamp, String comments, int severity, String duration)
    {
        super(type, timestamp, comments);
        this.severity = severity;
        this.duration = duration;
    }

    public int getSeverity()
    {
        return severity;
    }

    public void setSeverity(int severity)
    {
        this.severity = severity;
    }

    public String getDuration()
    {
        return duration;
    }

    public void setDuration(String duration)
    {
        this.duration = duration;
    }
}

File: mobile/src/main/java/net/stecky/puzzlemethis/events/behaviors/BadBehavior.kt
package net.stecky.puzzlemethis.events.behaviors

import net.stecky.puzzlemethis.Person
import net.stecky.puzzlemethis.events.EventSubType
import net.stecky.puzzlemethis.events.EventType
import net.stecky.puzzlemethis.events.notes.Note
import java.time.Period
import java.time.ZonedDateTime

/**
 * Created by sb185219 on 11/14/2017.
 */
class BadBehavior(subject: Person, author: Person, behaviorType: BehaviorSubType, timeStamp: ZonedDateTime,
                  notes: List<Note>, duration: Period, var severity: Int) :
        BehaviorBase(subject, author, EventType.BAD_BEHAVIOR, behaviorType, timeStamp, notes, duration)
{
}